lapis/src/main/kotlin/org/genspectrum/lapis/controller/LapisController.kt

```diff
@@ -894,6 +899,253 @@ class LapisController(
+        consumes = [MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_FORM_URLENCODED_VALUE],
+    )
+    @StringResponseOperation(
+        operationId = "postMostRecentCommonAncestorAsTsv",
+    )
+    fun postMostRecentCommonAncestorAsTsv(
+        @Parameter(schema = Schema(ref = "#/components/schemas/$MOST_RECENT_COMMON_ANCESTOR_RESPONSE_SCHEMA"))
+        @RequestBody
+        request: PhyloTreeSequenceFiltersRequestWithFields,
+        @RequestHeader httpHeaders: HttpHeaders,
+        response: HttpServletResponse,
+    ) {
+        lapisResponseStreamer.streamData(
+            request = request,
+            getData = ::getMostRecentCommonAncestorCollection,
+            response = response,
+            responseFormat = ResponseFormat.Csv(
+                delimiter = TAB,
+                acceptHeader = httpHeaders.accept,
+            ),
+        )
+    }
+
+    @GetMapping(MOST_RECENT_COMMON_ANCESTOR_ROUTE, produces = [MediaType.APPLICATION_JSON_VALUE])
+    @LapisMostRecentCommonAncestorResponse
+    fun getMostRecentCommonAncestorAsJson(
+        @PrimitiveFieldFilters
+        @RequestParam
+        sequenceFilters: GetRequestSequenceFilters?,
+        @PhyloTreeField
+        @RequestParam
+        phyloTreeField: String,
+        @Parameter(
+            description = PRINT_NODES_NOT_IN_TREE_FIELD_DESCRIPTION,
+        )
+        @RequestParam(required = false, defaultValue = "false")
+        printNodesNotInTree: Boolean,
+        @NucleotideMutations
+        @RequestParam
+        nucleotideMutations: List<NucleotideMutation>?,
+        @AminoAcidMutations
+        @RequestParam
+        aminoAcidMutations: List<AminoAcidMutation>?,
+        @DataFormat
+        @RequestParam
+        dataFormat: String? = null,
+        @NucleotideInsertions
+        @RequestParam
+        nucleotideInsertions: List<NucleotideInsertion>?,
+        @AminoAcidInsertions
+        @RequestParam
+        aminoAcidInsertions: List<AminoAcidInsertion>?,
+        response: HttpServletResponse,
+    ) {
+        val request = PhyloTreeSequenceFiltersRequestWithFields(
+            sequenceFilters?.filterKeys { !SPECIAL_REQUEST_PROPERTIES.contains(it) }
+                ?: emptyMap(),
+            nucleotideMutations ?: emptyList(),
+            aminoAcidMutations ?: emptyList(),
+            nucleotideInsertions ?: emptyList(),
+            aminoAcidInsertions ?: emptyList(),
+            phyloTreeField,
+            printNodesNotInTree = printNodesNotInTree,
+        )
+
+        lapisResponseStreamer.streamData(
+            request = request,
+            getData = ::getMostRecentCommonAncestorCollection,
+            response = response,
+            responseFormat = ResponseFormat.Json,
+        )
+    }
+
+    @GetMapping(MOST_RECENT_COMMON_ANCESTOR_ROUTE, produces = [TEXT_CSV_VALUE])
+    @StringResponseOperation(
+        operationId = "getMostRecentCommonAncestorAsCsv",
+    )
+    fun getMostRecentCommonAncestorAsCsv(
```

Comment:
Nitpick: Could you please reorder the methods? We usually have the GET methods before the POST methods